pkg/chunked/storage_linux.go

```diff
@@ -1753,13 +1756,20 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions, diff
 
 	var missingParts []missingPart
 
-	mergedEntries, totalSize, err := c.mergeTocEntries(c.fileType, toc.Entries)
+	mergedEntries, totalSizeFromTOC, err := c.mergeTocEntries(c.fileType, toc.Entries)
 	if err != nil {
 		return output, err
 	}
 
 	output.UIDs, output.GIDs = collectIDs(mergedEntries)
-	output.Size = totalSize
+	if convertedBlobSize > 0 {
+		// if the image was converted, store the original tar size, so that
+		// it can be recreated correctly.
+		output.Size = totalSizeFromTOC
+		output.Size = convertedBlobSize
```

Comment:
Is the second line overwriting the value that the first line just set?
(disclaimer: I don't know golang so well)

Reply:
yes, sorry.

Fixed now